### PR TITLE
[WIP] v3 add liquidity FE

### PR DESCRIPTION
**Pull request by ishwaryaa-b**

### CODE REVIEW

File: src/api/v3/helper.ts [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+
+  return {
+    address: tokenAddress,
+    tokenId: tokenId,
+    standard: tokenStandard,
+    decimals: tokenDecimals,
+  };
+};
+
+export const ContractStorage = async (
```
Comment (AnshuJalan) [inline src/api/v3/helper.ts:26]: small case first letter for function name

File: src/api/v3/helper.ts [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    let sqrtPriceFromLowerTick = Tick.computeSqrtPriceFromTick(lowerTick);
+    let sqrtPriceFromUpperTick = Tick.computeSqrtPriceFromTick(upperTick);
+
+    let liquidity = Liquidity.computeLiquidityFromAmount(
+      amount,
+      contractStorageParameters.sqrtPriceValue,
+      sqrtPriceFromLowerTick,
+      sqrtPriceFromUpperTick
+    );
+    console.log("---v3----", liquidity);
```
Comment (AnshuJalan) [inline src/api/v3/helper.ts:190]: remove log

File: src/api/v3/helper.ts [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+      lowerTickIndex: lowerTick,
+      upperTickIndex: upperTick,
+      lowerTickWitness: lowerTickWitness,
+      upperTickWitness: upperTickWitness,
+      liquidity: liquidity,
+      deadline: deadline,
+      maximumTokensContributed: maxTokenFinal,
+    };
+
+    let optionSet2 = {
```
Comment (AnshuJalan) [inline src/api/v3/helper.ts:255]: Remove this. Redundant variable

File: src/api/v3/liquidity.ts [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+export const calculateCurrentPrice = async (
+  tokenXSymbol: string,
+  tokenYSymbol: string,
+  refernceToken: String
+): Promise<any> => {
+  try {
+    let currentPrice;
+
+    let contractStorageParameters = await ContractStorage(tokenXSymbol, tokenYSymbol);
+    let PoolObject = new Pool(
```
Comment (AnshuJalan) [inline src/api/v3/liquidity.ts:18]: pool object is redundant, square root price is already there in contract storage

File: src/api/v3/liquidity.ts [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+      );
+    } else {
+      currentPrice = Price.computeRealPriceFromSqrtPrice(
+        contractStorageParameters.sqrtPriceValue,
+        contractStorageParameters.tokenX.decimals,
+        contractStorageParameters.tokenY.decimals
+      );
+    }
+
+    console.log(
```
Comment (AnshuJalan) [inline src/api/v3/liquidity.ts:43]: remove

File: src/api/v3/positions.ts [hunk 6]
```diff
@@ -0,0 +1,487 @@
...
+      };
+    });
+
+    return await Promise.all(finalPositionPromise);
+  } catch (error) {
+    console.log("v3 position error: ", error);
+  }
+};
+
+export const createIncreaseLiquidityOperation = async (
```
Comment (AnshuJalan) [inline src/api/v3/positions.ts:383]: Operation in API folder? @AniketSindhu

File: src/operations/v3/liquidity.ts [hunk 7]
```diff
@@ -0,0 +1,436 @@
...
+      success: false,
+      operationId: undefined,
+      error: error.message,
+    };
+  }
+};
+
+export const increaseLiquidity = async (
+  position: IV3PositionObject,
+  tokensAmount: BalanceNat, //
```
Comment (AnshuJalan) [inline src/operations/v3/liquidity.ts:201]: unused comment mark

**Review by AnshuJalan — CHANGES_REQUESTED** [review]
- Remove all console logs
- Function names must start with a small case
- Remove unused comments

File: src/api/v3/helper.ts [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+  } catch (error) {
+    console.log("v3 error: ", error);
+    return {
+      x: BigNumber("0"),
+      y: BigNumber("0"),
+    };
+  }
+};
+
+export const nearestTickIndex = async (): Promise<any> => {
```
[the remark this answers is not in the document]
Comment (AniketSindhu) [reply src/api/v3/helper.ts:110]: Removed

File: src/api/v3/positions.ts (hunk 6 above)
Comment (AniketSindhu) [reply src/api/v3/positions.ts:383]: This is not called from the frontend directly, The main operation is in the operation folder which internally calls this

File: src/components/v3/FeeTierMain.tsx [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    if (tokeninorg.name && tokenoutorg.name) {
+      checkPoolExistence(tokeninorg.name, tokenoutorg.name).then((res) => {
+        // if (tokeninorg.name != res.tokenA) {
+
+        // }
+        setIsExist(res);
+      });
+    }
+  }, [tokeninorg.name, tokenoutorg.name]);
+  console.log("isExist", isExist);
```
Comment (AniketSindhu) [inline src/components/v3/FeeTierMain.tsx:27]: remove console log

File: src/components/v3/FeeTierMain.tsx [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+    },
+    {
+      percentage: "0.3",
+      text: "Best for volatile pairs",
+      selectPercentage: "46",
+    },
+    {
+      percentage: "1",
+      text: "Best for exotic pairs",
+      selectPercentage: "46",
```
Comment (AniketSindhu) [inline src/components/v3/FeeTierMain.tsx:47]: Where is selectPercentage used?

File: src/components/v3/Increaseliq.tsx [hunk 11]
```diff
@@ -0,0 +1,432 @@
...
+  const [inputDisable, setInputDisable] = useState("false");
+  const [currentPrice, setCurrentPrice] = useState<BigNumber>(new BigNumber(0));
+  const tokensArray = Object.entries(tokens);
+  const tokeninorg = useAppSelector((state) => state.poolsv3.tokenInOrg);
+
+  const tokenoutorg = useAppSelector((state) => state.poolsv3.tokenOutOrg);
+
+  const tokensListConfig = useMemo(() => {
+    return tokensArray.map((token) => ({
+      name: token[0],
```
Comment (AniketSindhu) [inline src/components/v3/Increaseliq.tsx:58]: why is token[0] used in the name here and token[1] is used for image?

File: src/components/v3/LiquidityChartRangeInput/Brush.tsx [hunk 12]
```diff
@@ -0,0 +1,317 @@
...
+  innerHeight: number;
+  westHandleColor: string;
+  eastHandleColor: string;
+}) => {
+  const brushRef = useRef<SVGGElement | null>(null);
+  const brushBehavior = useRef<BrushBehavior<SVGGElement> | null>(null);
+  const topLevelSelectedToken = useAppSelector((state) => state.poolsv3.topLevelSelectedToken);
+  const tokenIn = useAppSelector((state) => state.poolsv3.tokenInOrg);
+  const tokenOut = useAppSelector((state) => state.poolsv3.tokenOut);
+  const tickSpacing = 10;
```
Comment (AniketSindhu) [inline src/components/v3/LiquidityChartRangeInput/Brush.tsx:60]: tick spacing hardcoded to 10 and I guess it is not being used anywhere else

File: src/components/v3/NewPoolV3/FeeTierNewPool.tsx [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+  feeTier: string;
+  isExist: any;
+}
+function FeeTierMainNewPool(props: IFeeTierMainProps) {
+  const fee = [
+    {
+      percentage: "0.01",
+      text: "Best for very stable pairs",
+      selectPercentage: "46",
+      created: true,
```
Comment (AniketSindhu) [inline src/components/v3/NewPoolV3/FeeTierNewPool.tsx:18]: created and selectPercentage hardcoded here.

File: src/components/v3/PriceRange.tsx [hunk 14]
```diff
@@ -0,0 +1,518 @@
...
+    tokensListConfig.map((tokenConfig) => {
+      if (tokenConfig.name === props.tokenIn.symbol) {
+        setTokenInConfig(tokenConfig.interface);
+      }
+      if (tokenConfig.name === props.tokenOut.symbol) {
+        setTokenOutConfig(tokenConfig.interface);
+      }
+    });
+  }, [tokensListConfig, props.tokenIn.symbol, props.tokenOut.symbol]);
+  const TickSpacing = (selectedFeeTier: string) => {
```
Comment (AniketSindhu) [inline src/components/v3/PriceRange.tsx:107]: function name should be in camelcase

**Review by AniketSindhu — CHANGES_REQUESTED** [review]

**Pull request closed without merging**